src/voting/CIP-voting.md

```diff
@@ -0,0 +1,168 @@
+
+```
+m / 1694' / 1815' / account' / role' / address_index'
+```
+
+`1694` (year Voltaire was born) Sets a dedicated `purpose` in the derivation path for the voting profile.  
+
+`address_index` - index of the key to use. 
+
+
+## **api.buildDelegation**(keys: GovernanceKey[], purpose: Purpose, networkId: number, stakeAccountPath: number = 0, stakeRolePath: number = 0, stakeIndex: number = 0): Promise\<**`Delegation`**>
+
+Should create the metadata object to be submitted by a metadata transaction to register the delegations on-chain. 
+
+### **Params**
+
+`account`: In case the wallet supports multiple accounts. defaults to 0
+
+`chain`: Part of the derivation path. defaults to 0
+
+`role`: 
+
+### **Returns**
+
+#### **`Delegation`**
+
+```
+export interface Delegation {
+    voting_delegation: GovernanceKey[],
+    staking_key: string,
+    reward_address: string,
+    nonce: number,
+    purpose: Purpose
+}
+```
+
+Defines the structure to be crafted and signed for delegation of voting & their respectively voting power.   Embeds the stake key and reward address from the wallet, and constructs a suitable nonce.
+
+***`voting_delegation`***: List of keys and their voting weight to delegate voting power to.
+
+The ***`staking_key`*** is Ed25519 public key 32bytes (x only) associated with the stake address. Defined in [CIP-11]((https://cips.cardano.org/cips/cip11), which specifies the derivation path for the staking key: 
+
+```
+m / 1852' / 1815' / account' / chain / 0
+```
+
+The ***`reward_address`*** as specified in [CIP-8](https://cips.cardano.org/cips/cip8/#addressformats) 
+
+
+The ***`nonce`*** is an unsigned integer (of CBOR major type 0) that should be monotonically increasing across all transactions with the same staking key. The wallet manages this and guarantees that nonces are always unique and greater than the previous ones. A suitable nonce value is the `linux epoch timestamp`.
+
+
+## **api.signDelegation**(delegation: DelegationMetadata, account: number = 0, role: number = 0, index: number = 0): Promise\<**`SignedDelegationMetadata`**>
+
+Since [CIP-18](https://cips.cardano.org/cips/cip18), multi-staking keys should be considered.  However, a single voting profile should exist per wallet. A single staking key should be used to perform EDDSA over the voting profile blake2b-256 hash.  The staking key used should still be the one defined in [CIP-11]((https://cips.cardano.org/cips/cip11). 
+
+### **Returns**
+
+#### **`SignedDelegationMetadata`**
+
+```
+export interface SignedDelegationMetadata {
+    '61284': DelegationMetadata,
+    '61285': string
+}
+```
+
+Defines the result of signing the DelegationMetadata.
+
+- `61284`: Key that defines the registration metadata map
+
+- `61285`: Signature of the blake2b hash of the `DelegationMetadata`
+
+## **api.submitMetadataTx(tx: MetadataTransaction): Promise\<hash32>**
+
+```
+export interface MetadataTransaction {
+  type: string,
```

Comment:
What is the idea behind this property? Would you ever need to build a voting transaction for era that is not the latest one? Wallet should know the era of the network it's connected to and how to build a transaction for it.